### PR TITLE
Add CodeQL scan workflow

### PR DESCRIPTION
**Pull request by omalyshe**
### Description 
Add CodeQL scan for the master branch. 
Try to use it for both cpp and python. If found as not feasible for cpp, we'll leave only for python. 

Fixes # - _issue number(s) if exists_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [x] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    # If this step fails, then you should remove it and run the build manually (see below)
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@4355270be187e1b672a7a1c7c7bae5afdc1ab94a # v3.24.10
+    
+    # If the analyze step fails for one of the languages you are analyzing with
+    # "We were unable to automatically build your code", modify the matrix above
+    # to set the build mode to "manual" for that language. Then modify this step
+    # to build your code.
+    # �� Command-line programs to run using the OS shell.
+    # � See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
```
Comment (isaevil) [inline .github/workflows/codeql.yml:72]: Something wrong with encoding?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["cpp", "python"]
+    
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@eb238b55efaa70779f274895e782ed17c84f2895 # v2.6.1
```
Comment (isaevil) [inline .github/workflows/codeql.yml:49]: Same comment about versions.

**Comment by omalyshe** [comment]
https://github.com/oneapi-src/oneTBB/actions/runs/9971386925

**Review by isaevil — APPROVED** [review]
LGTM

**Pull request merged**